### PR TITLE
Zig outline view

### PR DESCRIPTION
**Pull request by StringNick**
Release Notes:

- Added Zig outline view, related #7357 
- 
image

### CODE REVIEW

**Comment by StringNick** [comment]
@cla-bot check

**Comment by maxbrunsfeld** [comment]
It looks like there's a spelling typo in the highlight query. Could you take a look at that @StringNick ?

File: crates/zed/src/languages/zig/highlights.scm [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 
-parameter: (IDENTIFIER) @parameter
+;; func parametr
+parameter: (IDENTIFIER) @variable.parametr
 
 [
   field_member: (IDENTIFIER)
   field_access: (IDENTIFIER)
-] @field
+] @variable
```
Comment (maxbrunsfeld) [inline crates/zed/src/languages/zig/highlights.scm:22]: I think we usually use `@property` for struct fields / members. Would that make sense here?

**Review by maxbrunsfeld — CHANGES_REQUESTED** [review]

**Comment by maxbrunsfeld** [comment]
This is a great improvement. I left a couple suggestions.

**Comment by StringNick** [comment]
done

**Comment by maxbrunsfeld** [comment]
It looks like there's another typo in the highlights query.

**Comment by maxbrunsfeld** [comment]
There's still a typo being caught in CI. Can you run `typos` locally?

**Comment by maxbrunsfeld** [comment]
Thanks @StringNick !

**Pull request merged**